guibot/finder.py

```diff
@@ -104,6 +104,26 @@ def __repr__(self):
         return ("<value='%s' min='%s' max='%s' delta='%s' tolerance='%s' fixed='%s' enumerated='%s'>"
                 % (self.value, self.min_val, self.max_val, self.delta, self.tolerance, self.fixed, self.enumerated))
 
+    def __eq__(self, other):
+        """
+        Custom implementation for equality check.
+
+        :returns: whether this instance is equal to another
+        :rtype: bool
+        """
+        if not isinstance(other, CVParameter):
+            return NotImplemented
+        return repr(self) == repr(other)
+
+    def __ne__(self, other):
+        """
+        Custom implementation for inequality check.
+
+        :returns: whether this instance is different than another
+        :rtype: bool
+        """
+        return not self == other
```

Comment:
Isn't this the default `__ne__` implementation if we only define `__eq__` implementation here? If python implemented it in this way we would get this method for free.

Reply:
Indeed, it wasn't in [Python 2](https://docs.python.org/2/reference/datamodel.html#object.__eq__) ("when defining `__eq__()`, one should also define `__ne__()` so that the operators will behave as expected.") but this has changed in [Python 3](https://docs.python.org/3/reference/datamodel.html#object.__eq__) ("For `__ne__()`, by default it delegates to `__eq__()` and inverts the result unless it is `NotImplemented`.")